test/benchmarks/circuit_construction.py

```diff
@@ -93,4 +93,4 @@ def setup(self, width, gates, params):
         self.circuit, self.params = build_parameterized_circuit(width, gates, params)
 
     def time_bind_params(self, _, __, ___):
-        self.circuit.bind_parameters({x: 3.14 for x in self.params})
+        self.circuit.assign_parameters({x: 3.14 for x in self.params})
```

Comment:
We need to take a little care when modifying benchmarks, because any change invalidates the whole trace.  If we're doing this (and we probably should), we probably want to take a todo issue to write more complete benchmarks of `assign_parameters` that test more of the input formats / combinations.

Reply:
Yeah, it's right to change over the function immediately. We should probably remove the comment from the benchmark file and make it a GitHub issue instead - if we decided _not_ to change the benchmarks, we wouldn't be able to remove the comment without invalidating the history again haha!